js/query-parser.js

```diff
@@ -0,0 +1,118 @@
+		var prefilterValueParts = this.pvPreFilterQuery.value ? this.pvPreFilterQuery.value.split(',') : [];
+
+		if (
+			prefilterColumnParts.length !== prefilterLogicParts.length ||
+			prefilterLogicParts.length !== prefilterValueParts.length
+		) {
+			this.pvPreFilterQuery = null;
+			this.queryPreFilter = false;
+			console.warn('Please supply an equal number of parameter to the dynamicListPrefilter filters.');
+		} else {
+			this.pvPreFilterQuery = [];
+			var maxPartCount = Math.max(
+				prefilterColumnParts.length,
+				prefilterLogicParts.length,
+				prefilterValueParts.length
+			);
+			// loop through the query parts and create new filters with every one
+			for (let i = 0; i < maxPartCount; i++) {
+				var filter = {
+					column: prefilterColumnParts.pop(),
+					logic: prefilterLogicParts.pop(),
+					value: prefilterValueParts.pop()
+				};
+				this.pvPreFilterQuery.push(filter);
+			}
+		}
+	} else {
+		this.pvPreFilterQuery = null;
+	}
+
+	// dataSourceEntryId is always numeric
+	// we cast the one coming from query to a number
+	// so the equality check later passes
+	this.pvOpenQuery = _.pickBy({
+		id: parseInt(Fliplet.Navigate.query['dynamicListOpenId'], 10),
+		column: Fliplet.Navigate.query['dynamicListOpenColumn'],
+		value: Fliplet.Navigate.query['dynamicListOpenValue']
+	});
+	this.queryOpen = _(this.pvOpenQuery).size() > 0;
+	this.pvOpenQuery = this.queryOpen ? this.pvOpenQuery : null;
+
+	this.pvSearchQuery = _.pickBy({
+		column: Fliplet.Navigate.query['dynamicListSearchColumn'],
+		value: Fliplet.Navigate.query['dynamicListSearchValue']
+	});
+	this.querySearch = _(this.pvSearchQuery).size() > 0;
+	if (this.querySearch) {
+		// check if a comma separated list of columns were passed as column
+		var searchColumns = this.pvSearchQuery.column ? this.pvSearchQuery.column.split(',') : null;
+		this.pvSearchQuery.column = searchColumns && searchColumns.length ? searchColumns : this.pvSearchQuery.column;
+		this.data.searchEnabled = this.querySearch;
+	} else {
+		this.querySearch = null;
+	}
+
+	this.pvFilterQuery = _.pickBy({
+		value: Fliplet.Navigate.query['dynamicListFilterValue'],
+		hideControls: Fliplet.Navigate.query['dynamicListFilterHideControls']
+	});
+	this.queryFilter = _(this.pvFilterQuery).size() > 0;
+	if (this.queryFilter) {
+		// check if a comma separated list of columns were passed as column
+		var filterValues = this.pvFilterQuery.value ? this.pvFilterQuery.value.split(',') : null;
+		this.pvFilterQuery.value = filterValues && filterValues.length ? filterValues : this.pvFilterQuery.column;
+
+		// cast to boolean
+		(this.pvFilterQuery.hideControls = this.pvFilterQuery.hideControls
+			? this.pvFilterQuery.hideControls === 'true'
+			: false),
+			(this.data.filtersEnabled = this.queryFilter);
+	} else {
+		this.queryFilter = null;
+	}
+	return (
+		this.previousScreen ||
+		this.queryGoBack ||
+		this.queryPreFilter ||
+		this.queryOpen ||
+		this.querySearch ||
+		this.queryFilter
```

Comment:
I don't know if there's a right/wrong or a preferred way for Fliplet to do this (cc @squallstar ) but I usually go for

```js
		this.previousScreen
  		|| this.queryGoBack
  		|| this.queryPreFilter
  		|| this.queryOpen
  		|| this.querySearch
  		|| this.queryFilter
```

...with:

1. The operator in the beginning of the line
1. Indenting the subsequent lines